src/controllers/key.controller.ts

```diff
@@ -0,0 +1,51 @@
+import { Key } from "../models/key.model";
+import { keyStore } from "../store";
+import { ACTIONS } from "../types";
+
+export class KeyController<T extends HTMLElement> {
+  private readonly key: Key;
+  private readonly audio: HTMLAudioElement;
+  private readonly element: T;
+  private prevClickCount: number = 0;
+
+  constructor(key: Key, element: T) {
+    this.key = key;
+    this.element = element;
+    this.audio = new Audio(`/audio/${key.audio}.wav`);
+
+    keyStore.dispatch({ type: ACTIONS.ADD, payload: this.key });
+    keyStore.addListener((state) => this.updateComponent(state));
+  }
+
+  private updateComponent(state: Key[]) {
+    if (this.key.clickCount > this.prevClickCount) {
```

Comment:
This check might not be needed, as in our case, we have to update each key whenever some key is pressed right?

Reply:
ideally this playing sound must happen in the `handler` of each key press. Usually that is done in the view layer.